extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -116,6 +117,19 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Resolves relative URLs to their full path.
+   * @param {string} url
+   * @private
+   * @return {string}
+   */
+  resolveRelativeUrl_(url) {
```

Comment:
You don't need access to the class state. Export this as an individual function in a shared utility file.

```suggestion
export function resolveRelativeToSourceUrl(element, url) {
```

You'll have to pass `this.element` as `element`.